Learn/View Controllers/LessonConfigurationViewController.swift

```diff
@@ -0,0 +1,161 @@
+            cell.textLabel?.text = NSLocalizedString("Continue", comment: "Title of the button to begin lesson execution")
+
+            switch state {
+            case .editing:
+                cell.isEnabled = true
+                cell.isLoading = false
+            case .executing:
+                cell.isEnabled = false
+                cell.isLoading = true
+            }
+
+            return cell
+        } else {
+            return lesson.configurationSections[indexPath.section].cells[indexPath.item].tableView(tableView, cellForRowAt: indexPath)
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        if section < lesson.configurationSections.count {
+            return lesson.configurationSections[section].headerTitle
+        } else {
+            return nil
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        if section < lesson.configurationSections.count {
+            return lesson.configurationSections[section].footerTitle
+        } else {
+            return nil
+        }
+    }
+
+    // MARK: - UITableViewDelegate
+
+    override func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
+        return state == .editing
+    }
+
+    override func tableView(_ tableView: UITableView, willSelectRowAt indexPath: IndexPath) -> IndexPath? {
+        guard case .editing = state else {
+            return nil
+        }
+
+        tableView.endEditing(false)
+        tableView.beginUpdates()
+        hideDatePickerCells(excluding: indexPath)
+        return indexPath
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.endUpdates()
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        if indexPath.section == lesson.configurationSections.count {
+            state = .executing
+
+            lesson.execute { resultSections in 
+                dispatchPrecondition(condition: .onQueue(.main))
+
+                self.performSegue(withIdentifier: LessonResultsViewController.className, sender: resultSections)
+
+                self.state = .editing
+            }
+        }
+    }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        super.prepare(for: segue, sender: sender)
+
+        if let results = sender as? [LessonSectionProviding], let destination = segue.destination as? LessonResultsViewController {
+            destination.lesson = lesson
+            destination.results = results
+        }
+    }
+}
+
+
+extension LessonConfigurationViewController: DatePickerTableViewCellDelegate {
+    func datePickerTableViewCellDidUpdateDate(_ cell: DatePickerTableViewCell) {
+
+    }
+}
```

Comment:
The empty body of this method is perhaps an indicator that `hideDatePickerCells(excluding:)` doesn't need to be constrained to DatePickerTableViewCellDelegate implementors.
(If I were reading this and were unaware of the declaration of that function, I would've been confused by this protocol conformance, since the VC is never actually used as a delegate.)

Reply:
Yeah, it's a bit confusing. I'll see about getting this change into LoopKit